src/cci/cci_network.c

```diff
@@ -1010,85 +1096,92 @@ net_recv_stream (SOCKET sock_fd, unsigned char *ip_addr, int port, char *buf, in
 
   while (tot_read_len < size)
     {
-#if defined(WINDOWS)
-      FD_ZERO (&rfds);
-      FD_SET (sock_fd, &rfds);
-
-      if (timeout <= 0 || timeout > SOCKET_TIMEOUT)
+      if (con_handle->ssl_handle.is_connected == true && SSL_has_pending (con_handle->ssl_handle.ssl) > 0)
 	{
-	  tv.tv_sec = SOCKET_TIMEOUT / 1000;
-	  tv.tv_usec = (SOCKET_TIMEOUT % 1000) * 1000;
 	}
       else
```

Comment:
remove if-block and replace replace **`else`** as:
`if (con_handle->ssl_handle.is_connected == false || SSL_has_pending (con_handle->ssl_handle.ssl) <= 0)`